src/main/java/core/basesyntax/TreeSetCharacters.java

```diff
@@ -17,7 +23,37 @@
  * Результат 2: acf</p>
  */
 public class TreeSetCharacters {
-    public String getUniqueCharacters(String fileName) {
-        return null;
+    private static final int MAX_UNIQUE_CHARACTERS_QUANTITY = 5;
+
+    public String getUniqueCharacters(String fileName) throws IOException {
+        Set<Character> uniqueChars = new TreeSet<>();
+        try (BufferedReader input = new BufferedReader(new FileReader(fileName))) {
+            String nextString = null;
+            while (true) {
+                try {
+                    if ((nextString = input.readLine()) == null) {
+                        break;
+                    }
+                    String charsString = nextString.toLowerCase().replaceAll("[^a-z]", "");
+                    for (int i = 0; i < charsString.length(); i++) {
+                        uniqueChars.add(charsString.charAt(i));
+                    }
+                } catch (IOException e) {
+                    e.printStackTrace();
+                }
+            }
+        } catch (IOException e) {
+            throw e;
+        }
+
+        String result = "";
+        for (char c : uniqueChars) {
+            result += c;
```

Comment:
suggestion: better use string builder for this purpose because string is immutable and it always create new object after +=c